### PR TITLE
SPE flashlight fix

### PR DESCRIPTION
**Pull request by ante185**
## What type of PR is this.
1. [x] Bug
2. [ ] Change
3. [ ] Enhancement

### What have you changed and why?
Information:
Adds flashlights to the unit definitions
Fixes eventhandler inheritance by making I_G_Soldier_base_F inherit the EHs from SoldierGB correctly - this permits units derived from the vanilla FIA classes to use SPE flashlights, hipfire, some flamethrower hose stuff, SPE static weapons, fixes for the SPE alt reloads.

### Please specify which Issue this PR Resolves.
closes #XXXX

### Please verify the following and ensure all checks are completed.

1. [x] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [x] No
2. [ ] Yes (Please provide further detail below.)

### How can the changes be tested?
Steps: 

********************************************************
Notes:

### CODE REVIEW

**Review by Tiny-DM — APPROVED** [review]
lgtm

File: A3A/addons/core/CfgVehicles.hpp [hunk 1]
```diff
@@ -1,84 +1,95 @@
...
+        {
+            init = "if (local (_this select 0)) then {[(_this select 0), [], []] call BIS_fnc_unitHeadgear;};";
+            //init line to perserve the behaviour BI intended for the I_G_Soldier_base_F classs
+        };
+    };
     // Rebel AI unit types
     
     //don't need to change this one?
-    class I_G_Survivor_F;
+    class I_G_Survivor_F : I_G_Soldier_base_F{};
```
Comment (jaj22) [inline A3A/addons/core/CfgVehicles.hpp:17]: I tried removing these inheritance respecifications and I couldn't see any difference in the config tree. Maybe it'd matter if we were messing with the EventHandlers further in the subclasses, but that's not going to happen.

Comment (ante185) [reply A3A/addons/core/CfgVehicles.hpp:17]: Not sure if that is something we should do, my interpretation of https://community.bistudio.com/wiki/Class_Inheritance#Empty implies that we *shouldn't* remove the inheritance re-specification.
It should be to my knowledge safer to leave the re-specification as it doesn't actually change the inheritance.

Comment (jaj22) [reply A3A/addons/core/CfgVehicles.hpp:17]: The normal rules are these, I think:

// Just declares the existence. Does not change anything about that class, including the inheritance
`class I_G_Survivor_F;`

// Removes the previous inheritance but not the properties?
`class I_G_Survivor_F {};`

// (Re)sets the inheritance to I_G_Soldier_base_F
`class I_G_Survivor_F : I_G_Soldier_base_F {};`

Note that if you create a child class and then change a value in a parent class later, the child class inherits the new value. You could flip this file around and do the SoldierGB inheritance patch later, and it'd still work.

Doesn't really matter either way though. If it worries you then we can keep it.

**Pull request merged**